src/plugins/data/common/search/tabify/response_writer.ts

```diff
@@ -50,15 +50,25 @@ export class TabbedAggResponseWriter {
   row() {
     const rowBuffer: TabbedAggRow = {};
 
-    this.bucketBuffer.forEach((bucket) => {
+    for (let i = 0; i < this.bucketBuffer.length; i++) {
+      const bucket = this.bucketBuffer[i];
```

Comment:
Why not use the equivalent `for of`?

```suggestion
    for (const bucket of this.bucketBuffer) {
```

I find the `for of` version clearer to read, and as far as I can see they have the same equivalent allocation footprint.
If the are concerns about the actual compiler output, I've checked in the TS playground and essentially the code remains unchanged unless the target is ES5, where it is transpiled into this version anyway.

Reply:
In the spirit of the PR, I would keep it as a regular loop. `for ... of` under the hood creates a new iterator. 

The creation has quite a complex algorithm.

- https://stackoverflow.com/questions/54171113/how-is-for-of-implemented-in-js
- https://tc39.es/ecma262/#sec-createarrayiterator

It allocates new iterator objects too, and allocation in tight loops is what we are trying to avoid with this PR.

Overall, `for ..of` is the slowest loop because of it:
- https://dev.to/siddiqus/which-for-loop-is-the-fastest-in-javascript-4hdf

`BucketBuffer` is also explicitly typed as an array (https://github.com/elastic/kibana/blob/a2438333ba136fba3763eb4c5af90fadcfca3ba4/src/plugins/data/common/search/tabify/response_writer.ts#L28), so it seems appropriate to loop over it with the most vanilla of loops.